pkgs/native_assets_cli/lib/src/code_assets/config.dart

```diff
@@ -47,37 +47,55 @@ class CodeConfig {
   /// The operating system being compiled for.
   final OS targetOS;
 
-  late final IOSConfig? _iOSConfig;
-  late final AndroidConfig? _androidConfig;
-  late final MacOSConfig? _macOSConfig;
-
-  CodeConfig(HookConfig config)
-      : linkModePreference = LinkModePreference.fromString(
-            config.json.string(_linkModePreferenceKey)),
-        // ignore: deprecated_member_use_from_same_package
-        _targetArchitecture = (config is BuildConfig && config.dryRun)
-            ? null
-            : Architecture.fromString(config.json.string(_targetArchitectureKey,
-                validValues: Architecture.values.map((a) => a.name))),
-        targetOS = OS.fromString(config.json.string(_targetOSConfigKey)),
-        cCompiler = switch (config.json.optionalMap(_compilerKey)) {
-          final Map<String, Object?> map => CCompilerConfig.fromJson(map),
-          null => null,
-        } {
-    // ignore: deprecated_member_use_from_same_package
-    _iOSConfig = (config is BuildConfig && config.dryRun) || targetOS != OS.iOS
+  final IOSConfig? _iOSConfig;
+  final AndroidConfig? _androidConfig;
+  final MacOSConfig? _macOSConfig;
+
+  CodeConfig._({
```

Comment:
Consider keeping the old code here and renaming this to
```
// TODO: Merge with `.fromJson` once dry-run is gone.
CodeConfig._fromJsonAndDryRun(Map<String, Object?> json, bool dryRun) {}

factory CodeConfig.fromJson(Map<String, Object?) json) {
  return CodeConfig._fromJsonAndDryRun(json, json.getOptional<bool>(_dryRunConfigKey) ?? false);
}
```

Otherwise there's high chance someone may want to make it public in the future.

Reply:
> `Map<String, Object?> json, bool dryRun`

There is no nesting currently, so `dryRun` is in the same root level as the code config fields.

> Otherwise there's high chance someone may want to make it public in the future.

Left a comment to avoid someone doing this.